src/OpenLoco/Windows/PromptBrowseWindow.cpp

```diff
@@ -985,11 +1021,37 @@ namespace OpenLoco::Ui::Windows::PromptBrowse
     }
 
     // 0x00446E87
-    // TODO: only called by this window -- implement.
-    static void sub_446E87(Window* self)
+    static void loadFileHeader(Window* self)
     {
-        registers regs;
-        regs.esi = X86Pointer(self);
-        call(0x00446E87, regs);
+        freeFileHeader();
+
+        _9DA285 = 0;
+        if (self->var_85A == -1)
+            return;
+
+        file_entry& entry = _files[self->var_85A];
+        if (entry.is_directory())
+            return;
+
+        // Create full path to target file.
+        fs::path path = fs::u8path(&_directory[0]) / std::string(entry.get_name());
```

Comment:
`get_name` function should really not do any processing its already a stem. Something for future to change.